### PR TITLE
Use case show courses

### PR DESCRIPTION
**Pull request by vivic04**

### CODE REVIEW

**Review by 999Vik — CHANGES_REQUESTED** [review]
Pull from main and re-request

File: src/main/java/relay/interface_adapter/show_courses/ShowCoursePresenter.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+    // Method from the ShowCourseOutputBoundary interface
+    @Override
+    public ResponseEntity<Map<String, Object>> prepareSuccessResponce(ShowCourseOutputData outputData) {
+        // Extract relevant data from the output and update the ViewModel
+        ShowCourseState showCourseState = (ShowCourseState) viewModel.getState();
+        List<String> courseNames = outputData.getCourses();
+        showCourseState.setCourses(courseNames);
+        showCourseState.setStatusCode(HttpStatus.CREATED);
```
Comment (Ram-Raghav-S) [inline src/main/java/relay/interface_adapter/show_courses/ShowCoursePresenter.java:29]: Change the success status code to OK instead of CREATED.

**Review by Ram-Raghav-S — CHANGES_REQUESTED** [review]
Looks good apart from a few changes. We will need to add Java doc and unit tests, but we can leave that for later. For now, we can merge in it after fixing the changes.

**Review by Ram-Raghav-S — APPROVED** [review]
Looks good. Thanks for making the changes.

**Pull request merged**